src/Nethermind/Nethermind.Synchronization/BeamSync/BeamBlockchainProcessor.cs

```diff
@@ -260,18 +261,26 @@ private void BeamProcess(Block block)
                 _recoveryStep.RecoverData(block);
                 (IBlockchainProcessor beamProcessor, IStateReader stateReader) = CreateProcessor(block, new ReadOnlyDbProvider(_readOnlyDbProvider, true), _specProvider, _logManager);
 
-                BlockHeader parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
+                BlockHeader? parentHeader = _readOnlyBlockTree.FindHeader(block.ParentHash, BlockTreeLookupOptions.TotalDifficultyNotNeeded);
                 if (parentHeader != null)
                 {
-                    prefetchTasks = PrefetchNew(stateReader, block, parentHeader.StateRoot, parentHeader.Author ?? parentHeader.Beneficiary);
+                    if (parentHeader.StateRoot == null)
+                    {
+                        throw new InvalidDataException($"Received a block with null {nameof(parentHeader.StateRoot)} for beam processing");
+                    }
+                    if (parentHeader.GasBeneficiary == null)
+                    {
+                        throw new InvalidDataException($"Received a block with null {nameof(parentHeader.GasBeneficiary)} for beam processing");
+                    }
+                    prefetchTasks = PrefetchNew(stateReader, block, parentHeader.StateRoot, parentHeader.GasBeneficiary);
                 }
                 
                 Stopwatch stopwatch = Stopwatch.StartNew();
                 Block? processedBlock = null;
                 beamProcessingTask = Task.Run(() =>
                 {
                     BeamSyncContext.MinimumDifficulty.Value = block.TotalDifficulty.Value;
-                    BeamSyncContext.Description.Value = $"[preProcess of {block.Hash.ToShortString()}]";
+                    BeamSyncContext.Description.Value = $"[preProcess of {block.Hash?.ToShortString() ?? "<block with null hash>"}]";
```

Comment:
I wasn't sure what to do here.  Throwing just because a log line couldn't be written felt wrong, but that is what this code would have done (NRE).  Instead, just putting a note felt like it would make the issue obvious to the reader but let the system proceed (since it looks like it is resilient to block processing failure anyway).

Reply:
makes sense